src/var.c

```diff
@@ -250,11 +271,70 @@ static int fi_parse_bool(const char *str_value)
 	return -1;
 }
 
+static void load_conf(void)
+{
+	char *tmp;
+	struct ofi_conf_entry *conf;
+	size_t line_len = 0;
+	char line[MAX_CONF_LINE_LENGTH];
+
+	dlist_init(&conf_list);
+
+	FILE *fptr = fopen(DEFAULT_CONF_FILE_PATH, "r");
+	if (!fptr)
+		return;
+
+	while (fgets(line, MAX_CONF_LINE_LENGTH, fptr) != NULL) {
+		tmp = strchr(line, '=');
+		if (!tmp)
+			continue;
+
+		line_len = strlen(line);
+		tmp[0] = '\0';
+		if (line[line_len-1] == '\n')
+			line[line_len-1] = '\0';
+
+		conf = calloc(1, sizeof(*conf));
+		if (!conf)
+			break;
+
+		conf->name = strdup(line);
+		if (!conf->name) {
+			free_conf(conf);
+			break;
+		}
+
+		conf->value = strdup(&tmp[1]);
+		if (!conf->value) {
+			free_conf(conf);
+			break;
+		}
+
+		dlist_insert_tail(&conf->entry, &conf_list);
```

Comment:
It would be nice to add FI_INFO() here using the core to print all read name-value pairs.  This can be added as a separate change.

Reply:
the logging interface requires an instance of a provider, so we can either dump this late (maybe during getinfo?) or we need to add a logging function with that requirement removed.

Reply:
ah, tmyk... 

Other issue is that this takes place prior to fi_log_init, so the logging system hasn't been initialized. The logging system relies on these variables, so it seems a catch 22